paddle/framework/.clang-format

```diff
@@ -0,0 +1,5 @@
+---
+Language:        Cpp
```

Comment:
/.clang-format was created by @reyoung to fit existing code style.  However, we'd like to follow strict Google C++ style, so that during code review we could paste links into Google style's online doucment to help use explain pros and cons of programming choices. 

clang-format works by looking for .clang-format in the nearest parent directory of the source file.  So, as we put the new .clang-format in paddle/framework, clang-format would check paddle/framework/ddim.h and other source files following Google C++ style.